src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1042,10 +1042,67 @@ Node* VectorNode::try_to_gen_masked_vector(PhaseGVN* gvn, Node* node, const Type
   }
 }
 
+bool VectorNode::can_swap_inputs() {
+  // Must be a binary operation.
+  if (req() != 3) {
+    return false;
+  }
+
+  // Predicated vector operations are sensitive to ordering of inputs.
+  // When the mask corresponding to a vector lane is false then
+  // the result of the operation is corresponding lane of its first operand.
+  //   i.e. RES = VEC1.lanewise(OPER, VEC2, MASK) is semantically equivalent to
+  //        RES = VEC1.BLEND(VEC1.lanewise(OPER, VEC2), MASK)
+  // Here, first operand "VEC1" represents both source and destination vector.
```

Comment:
Sorry @jatin-bhateja . But this still does not make sense to me. I thought the `RES` was the destination vector for the `BLEND`? Does `BLEND` not have 3 inputs (2 vectors + 1 mask vector), and one output vector?

So I would expect a `BLEND` to have a form like this:
`RES = BLEND(V1, V2, M)`

Reply:
Reframed for clarity.